File: BinaryTree/php/treeLevels.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/TreeNode.php';

function treeLevels(?TreeNode $root): array
{
    if ($root === null) {
        return [];
    }

    $queue = [[ $root, 0 ]];
    $levels = [];

    while ($queue) {
        [$node, $level] = array_shift($queue);
        if (!array_key_exists($level, $levels)) {
            $levels[$level] = [];
        }
        $levels[$level][] = $node->val;
        if ($node->left !== null) {
            $queue[] = [$node->left, $level + 1];
        }
        if ($node->right !== null) {
            $queue[] = [$node->right, $level + 1];
        }
    }

    ksort($levels);
    return array_values($levels);
}

File: BinaryTree/java/treeLevels.java
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.List;
import java.util.Queue;

class TreeLevels {
  public static List<List<Integer>> treeLevels(TreeNode root) {
    List<List<Integer>> levels = new ArrayList<>();
    if (root == null) {
      return levels;
    }

    Queue<TreeNode> queue = new ArrayDeque<>();
    queue.add(root);

    while (!queue.isEmpty()) {
      int levelSize = queue.size();
      List<Integer> level = new ArrayList<>();
      for (int i = 0; i < levelSize; i += 1) {
        TreeNode current = queue.remove();
        level.add(current.val);
        if (current.left != null) queue.add(current.left);
        if (current.right != null) queue.add(current.right);
      }
      levels.add(level);
    }

    return levels;
  }

  public static void run() {
    // intentionally empty
  }
}
